datadog/resource_datadog_dashboard.go

```diff
@@ -4722,6 +4722,257 @@ func buildTerraformTimeseriesDefinition(datadogDefinition datadogV1.TimeseriesWi
+							},
+							"name": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "Name of the query for use in formulas.",
+							},
+						},
+					},
+				},
+				"event_query": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					MaxItems:    1,
+					Description: "A timeseries formula and functions events query.",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"data_source": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "Data source for event platform-based queries.",
+							},
+							"search": {
+								Type:        schema.TypeList,
+								Optional:    true,
+								MaxItems:    1,
+								Description: "Search options.",
+								Elem: &schema.Resource{
+									Schema: map[string]*schema.Schema{
+										"query": {
+											Type:         schema.TypeString,
+											ValidateFunc: validators.ValidateStringValue,
+											Required:     true,
+											Description:  "Events search string.",
+										},
+									},
+								},
+							},
+							"indexes": {
+								Type:        schema.TypeList,
+								Optional:    true,
+								Elem:        &schema.Schema{Type: schema.TypeString},
+								Description: "An array of index names to query in the stream.",
+							},
+							"compute": {
+								Type:        schema.TypeList,
+								Required:    true,
+								Description: "Compute options.",
+								Elem: &schema.Resource{
+									Schema: map[string]*schema.Schema{
+										"aggregation": {
+											Type:        schema.TypeString,
+											Required:    true,
+											Description: "Aggregation methods for event platform queries.",
+										},
+										"interval": {
+											Type:        schema.TypeInt,
+											Optional:    true,
+											Description: "A time interval in milliseconds.",
+										},
+										"metric": {
+											Type:        schema.TypeString,
+											Optional:    true,
+											Description: "Measurable attribute to compute.",
+										},
+									},
+								},
+							},
+							"group_by": {
+								Type:        schema.TypeList,
+								Optional:    true,
+								Description: "Group by options.",
+								Elem: &schema.Resource{
+									Schema: map[string]*schema.Schema{
+										"facet": {
+											Type:        schema.TypeString,
+											Required:    true,
+											Description: "Event facet.",
+										},
+										"limit": {
+											Type:        schema.TypeInt,
+											Optional:    true,
+											Description: "Number of groups to return.",
+										},
+										"sort": {
+											Type:        schema.TypeList,
+											Optional:    true,
+											MaxItems:    1,
+											Description: "Options for sorting group by results.",
+											Elem: &schema.Resource{
+												Schema: map[string]*schema.Schema{
+													"aggregation": {
```

Comment:
I'm a bit confused by this schema. What does it mean to sort by aggregation=pc99 for example? Shouldn't we just sort by aggregation?

If it's intended, it's missing the enum validation.

Reply:
It is indeed intended [source](https://github.com/DataDog/dogweb/blob/25252010da5443871e7814fddbc098703e265f86/dogweb/lib/validation/schemas/query.py#L126). I'll add an enum validation.

Reply:
Done in [commit 2f914c9](https://github.com/DataDog/terraform-provider-datadog/pull/892/commits/2f914c9da8e833c9a46c5c24bd654aa47ba44918)